Client/Application/AppDelegate.swift

```diff
@@ -294,6 +368,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
                 url = item.value
             case "private":
                 isPrivate = NSString(string: item.value ?? "false").boolValue
+            case "deep-link":
+                guard let value = item.value else { break }
+                Router.shared.routeURL(value)
```

Comment:
I agree - the delegate is likely called on the main thread. I'm happy to keep the assert and we will just find out if our assumption is wrong during testing _hopefully_. I don't see anything in the docs on what thread it is called on!